larsim/LegacyLArG4/ParticleListAction.h

```diff
@@ -105,6 +109,9 @@ namespace larg4 {
     std::map<int, GeneratedParticleIndex_t> const& GetPrimaryTruthMap() const
       { return fPrimaryTruthMap; }
 
+    /// Returns whether a particle list is being kept.
+    bool hasList() const { return bool(fparticleList); }
```

Comment:
Recommend:
```.cpp
bool hasList() const noexcept { return static_cast<bool>(fparticleList); }
```

Reply:
I miss something more descriptive...

Reply:
Two issues:

- C-style casts are are discouraged
- The `std::unique<T>::operator bool()` member function is guaranteed to not throw an exception, therefore the `noexcept` keyword.

Reply:
Sorry, I meant that I miss an interface element of `std::unique_ptr` more descriptive than a cast to a different type.
Like `cet::exempt_ptr::empty()`, for example.

Reply:
Commit e554efd.